pkg/proxy/apis/kubeproxyconfig/scheme/scheme.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheme
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/kubernetes/pkg/proxy/apis/kubeproxyconfig"
+	"k8s.io/kubernetes/pkg/proxy/apis/kubeproxyconfig/v1alpha1"
+)
+
+var (
+	// Scheme defines methods for serializing and deserializing API objects.
+	Scheme = runtime.NewScheme()
+	// Codecs provides methods for retrieving codecs and serializers for specific
+	// versions and content types.
+	Codecs = serializer.NewCodecFactory(Scheme)
+)
+
+func init() {
+	v1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
```

Comment:
The 2nd argument needs to be `v1alpha1.SchemeGroupVersion`. Right @deads2k @sttts ?

Reply:
@deads2k @sttts @xialonglee sig-network has 2 PRs that are on hold waiting for this to merge. We can really use your help coming to a resolution.

Reply:
> The 2nd argument needs to be v1alpha1.SchemeGroupVersion. Right @deads2k @sttts ?

It depends on how you want to serialize and deserialize them.  Most external versions already do this registration inside their own group to serialize out natively.  If you need to support this for interoperability it is ok.

Reply:
> The 2nd argument needs to be v1alpha1.SchemeGroupVersion. Right @deads2k @sttts ?
> It depends on how you want to serialize and deserialize them. Most external versions already do this registration inside their own group to serialize out natively. If you need to support this for interoperability it is ok.

I should also say that if just cargo-culted it (I'm not judging), I'd remove it until you had proof it was required for something like deserializing some already serialized data you have to be compatible with.

Reply:
I'm going to guess it was copy/paste. So you recommend removing it?

Reply:
>I'm going to guess it was copy/paste.

yeah... :sweat_smile: So should I change this to `v1.AddToGroupVersion(Scheme, v1alpha1.SchemeGroupVersion`, right? TBH, I didn't understand more details here :)

Reply:
@xiangpengzhao can you remove this line entirely and make sure loading the configuration file still works?

Reply:
@ncdc I will verify this later :)

Reply:
removing this is still outstanding, right?

Reply:
If it doesn't break anything, I'd rather do it in a follow-up. This has been waiting long enough.